feat: use zero trust dialer again

dovholuknf opened the pull request:
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?)
- [ ] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

## Testing Instructions
<!-- How can the reviewers test your change? -->

## New Dependency Instructions (If applicable)
<!-- Please follow [vetting instructions](https://wiki.edgexfoundry.org/display/FA/Vetting+Process+for+3rd+Party+Dependencies) and place results here -->

cloudxxx8 commented on bootstrap/registration/registry.go:
```diff
@@ -69,10 +68,10 @@ func createRegistryClient(
 		// Bypass the zero trust zitidfied transport for Core Keeper Registry client
 		// Should leverage the HttpTransportFromService function from zerotrust pkg, set the default transport for now
-		secretProvider.SetHttpTransport(http.DefaultTransport)
+		// secretProvider.SetHttpTransport(http.DefaultTransport)
```
```
if (bootstrapConfig.Registry.Type == "keeper") {
  secretProvider.SetHttpTransport(http.DefaultTransport)
}
```

dovholuknf replied:
updated

cloudxxx8 reviewed (approved):
LGTM

dovholuknf commented:
@cloudxxx8 -- i removed a second location where the bypass was put in place. i didn't see and equivalent of:

> bootstrapConfig.Registry.Type == "keeper"

so i removed it. Maybe I missed it?

cloudxxx8 commented:
close this PR due to the byPass code is removed from https://github.com/edgexfoundry/go-mod-bootstrap/pull/736
